java/src/main/native/src/row_conversion.cu

```diff
@@ -1636,19 +1660,36 @@ std::vector<std::unique_ptr<column>> convert_to_rows(table_view const &tbl,
       util::div_rounding_up_unsafe(validity_tile_infos.size(), NUM_VALIDITY_TILES_PER_KERNEL));
   dim3 validity_threads(std::min(validity_tile_infos.size() * 32, 128lu));
 
-  detail::row_offset_functor offset_functor(fixed_width_size_per_row);
+  if (fixed_width_only) {
+    detail::row_offset_functor offset_functor(
+        util::round_up_unsafe(fixed_width_size_per_row, JCUDF_ROW_ALIGNMENT));
+
+    detail::copy_to_rows<<<blocks, threads, total_shmem_in_bytes, stream.value()>>>(
+        num_rows, num_columns, shmem_limit_per_tile, gpu_tile_infos, dev_input_data.data(),
+        dev_col_sizes.data(), dev_col_starts.data(), offset_functor,
+        batch_info.d_batch_row_boundaries.data(),
+        reinterpret_cast<int8_t **>(dev_output_data.data()));
+
+    detail::copy_validity_to_rows<<<validity_blocks, validity_threads, total_shmem_in_bytes,
+                                    stream.value()>>>(
+        num_rows, num_columns, shmem_limit_per_tile, offset_functor,
+        batch_info.d_batch_row_boundaries.data(), dev_output_data.data(), column_starts.back(),
+        dev_validity_tile_infos, dev_input_nm.data());
+  } else {
+    detail::string_row_offset_functor offset_functor(batch_info.batch_row_offsets);
```

Comment:
Seems like these two functors could also be passed in as template parameters, eliminating the need for this if statement.

Reply:
Would that not just push the if higher up in the callstack to determine which iterator needs instantiation?

In the next iteration of this code(see #10235 ) this if/else grows to include more calculations and another kernel launch for the string data. It isn't just a question of what iterator to use, but if we can encapsulate it in some way I am all for it.

Reply:
I was able to do this by pulling out the `batch_info` creation, since it is needed for the string iterator. I think it cleans this up nicely as it also allows me to not pass the templated iterator.